tirocks/src/properties.rs

```diff
@@ -0,0 +1,556 @@
+}
+
+define! {
+    int
+    /// Number of level to which L0 data will be compacted.
+    PropBaseLevel, crocksdb_property_name_base_level
+}
+
+define! {
+    int
+    /// Estimated total number of bytes compaction needs to rewrite to get all levels down
+    /// to under target size. Not valid for other compactions than level-based.
+    PropEstimatePendingCompactionBytes, crocksdb_property_name_estimate_pending_compaction_bytes
+}
+
+define! {
+    /// A string representation of the aggregated table properties of the target column family.
+    PropAggregatedTableProperties, crocksdb_property_name_aggregated_table_properties
+}
+
+/// Same as the [`PropAggregatedTableProperties`] but only returns the aggregated table
+/// properties of the pecified level at the target column family.
+pub struct PropAggregatedTablePropertiesAtLevel {
+    key: Vec<u8>,
+}
+
+impl PropAggregatedTablePropertiesAtLevel {
+    pub fn new(level: usize) -> Self {
+        let prefix = unsafe {
+            let mut buf = r(&[]);
+            tirocks_sys::crocksdb_property_name_aggregated_table_properties_at_level(&mut buf);
+            s(buf)
+        };
+        let mut key = Vec::with_capacity(prefix.len() + 1);
+        key.extend_from_slice(prefix);
+        write!(key, "{}", level).unwrap();
+        Self { key }
+    }
+}
+
+impl Property for PropAggregatedTablePropertiesAtLevel {
+    #[inline]
+    fn key(&self) -> &[u8] {
+        &self.key
+    }
+}
+
+define! {
+    int
+    /// The current actual delayed write rate. 0 means no delay.
+    PropActualDelayedWriteRate, crocksdb_property_name_actual_delayed_write_rate
+}
+
+define! {
+    int
+    /// 1 if write has been stopped.
+    PropIsWriteStopped, crocksdb_property_name_is_write_stopped
+}
+
+define! {
+    int
+    /// 1 if write has been stalled.
+    PropIsWriteStalled, crocksdb_property_name_is_write_stalled
+}
+
+define! {
+    int
+    /// An estimation of oldest key timestamp in the DB. Currently only available for
+    /// FIFO compaction with compaction_options_fifo.allow_compaction = false.
+    PropEstimateOldestKeyTime, crocksdb_property_name_estimate_oldest_key_time
+}
+
+define! {
+    int
+    /// Block cache capacity.
+    PropBlockCacheCapacity, crocksdb_property_name_block_cache_capacity
+}
+
+define! {
+    int
+    /// The memory size for the entries residing in block cache.
+    PropBlockCacheUsage, crocksdb_property_name_block_cache_usage
+}
+
+define! {
+    int
+    /// The memory size for the entries being pinned.
+    PropBlockCachePinnedUsage, crocksdb_property_name_block_cache_pinned_usage
+}
+
+define! {
+    /// Multi-line string of options.statistics.
+    PropOptionsStatistics, crocksdb_property_name_options_statistics
+}
+
+pub struct PropertyMap {
+    ptr: *mut crocksdb_map_property_t,
+}
+
+impl Drop for PropertyMap {
+    #[inline]
+    fn drop(&mut self) {
+        unsafe { tirocks_sys::crocksdb_destroy_map_property(self.ptr) }
```

Comment:
Naming is inconsistent:
```suggestion
        unsafe { tirocks_sys::crocksdb_map_property_destroy(self.ptr) }
```